src/rust/engine/src/nodes.rs

```diff
@@ -397,18 +487,21 @@ pub struct SelectDependencies {
   pub subject: Key,
   pub variants: Variants,
   pub selector: selectors::SelectDependencies,
+  pub edges: rule_graph::RuleEdges
```

Comment:
As I said above, I don't need to keep them in Select* nodes. I'll do that replace before landing this. That should keep them pretty small since they'll only include entries that match up to two of the relevant Selects instead of all of the ones for the Node. Would that satisfy your concern in the short term?